deerlab/snlls.py

```diff
@@ -247,9 +247,9 @@ def snlls(y, Amodel, par0, lb=None, ub=None, lbl=None, ubl=None, nnlsSolver='cvx
     nonNegativeOnly = (np.all(lbl == 0)) and (np.all(np.isinf(ubl)))
 
 
+    # Use an arbitrary axis
+    ax = np.arange(1, Nlin+1)
     if includePenalty:

```

Comment:
Why do you set an L even if includePenalty is False?

Reply:
An arbitrary `L` is defined when `includePenalty=False` to avoid a reference-error later in line 303
```python
AtA, Aty = dl.lsqcomponents(y, A, L, alpha, weights, regtype=regtype)
````

Reply:
That's not clean. I think in the absence of a reg term, `lsqcomponents` should only be called wity `y` and `A`.

Reply:
See next comment.